src/components/Header/index.js

```diff
@@ -1,140 +1,76 @@
-import { useRef, useState } from 'react';
-import AppBar from '@mui/material/AppBar';
-import Toolbar from '@mui/material/Toolbar';
+import * as React from "react";
+import { useState, useEffect } from 'react';
+import { Link, useMatch, useResolvedPath } from 'react-router-dom';
 import IconButton from '@mui/material/IconButton';
-import Box from '@mui/material/Box';
-import Stack from '@mui/material/Stack';
 import * as Icons from 'components/Shared/Icons';
-import * as Styled from './styles';
-import { useEventListener } from 'hooks';
 import SearchModal from './SearchModal';
+import './navbar.scss';
 
-const Header = () => {
-  const navLinksRef = useRef();
+function CustomLink({ children, to, ...props }) {
+  const resolved = useResolvedPath(to);
+  const match = useMatch({ path: resolved.pathname, end: true });
 
-  const [openSearch, setOpenSearch] = useState(false);
+  return (
+    <Link
+      className={`router-link ${match ? "router-link-active" : ""}`}
+      to={to}
+      {...props}
+    >
+      {children}
+    </Link>
+  );
+}
 
-  const handleOpen = () => setOpenSearch(true);
+const Header = () => {
+  const [openSearch, setOpenSearch] = useState(false);
+  const [navbarOpen, setNavbarOpen] = useState(false);
+  const [navbarClosing, setNavbarClosing] = useState(false);
 
-  useEventListener(
-    'click',
-    event => {
-      const { target, currentTarget } = event;
-      const navLinks = currentTarget.children;
-      const [active] = [...navLinks].filter(link => link.classList.contains('active'));
-      if (active && active !== target) {
-        active.classList.remove('active');
-        target.classList.add('active');
-      }
-    },
-    navLinksRef.current,
-  );
+  useEffect(() => {
+    setNavbarClosing(true);
+    const timer = setTimeout(() => {
+      setNavbarClosing(false);
+    }, 650);
```

Comment:
If these flags are meant for animating the nav-bar (which I believe they are) it would make sense to wrap them into the `requestAnimationFrame` method to avoid jank.
```suggestion
    requestAnimationFrame(() => setNavbarClosing(true));
    const timer = setTimeout(() => {
      requestAnimationFrame(() => setNavbarClosing(false));
    }, 650);
```